internal/watcher/watcher.go

```diff
@@ -31,8 +30,12 @@ type Watcher interface {
 	// Implementors must not return a nil function.
 	Watch(ctx context.Context) (*BlobEvent, Cleanup, error)
 
+	// Download copies the file or directory identified by key to dest.
+	Download(ctx context.Context, dest, key string) error
```

Comment:
I wish that a `Watcher` didn't have to know how to download, or maybe that's ok? It sounds like something that we should be able to do using the `*blob.Bucket` returned by `OpenBucket` in the service layer, but I don't know if that's possible. `*blob.Bucket` was recently updated to implement the standard io/fs.FS and io/fs.SubFS interfaces (https://github.com/google/go-cloud/pull/3272) so you should be able to walk a bucket with [io/fs.WalkDir](https://pkg.go.dev/io/fs#WalkDir), but I don't know if that's enough. Maybe BlobEvent.IsDir() is also relevant. There's probably some room for improvement, but I'm happy with what you've done here. The internal/watcher package has always been at the back of my mind as something not very well-conceived, rushed and disorganized.

Reply:
I agree that it would be better if the Watcher service was solely concerned with watching for transfers and emitting change events, with downloading the blob the responsibility of a separate module - but I think this would be a big change. I'll file an issue about separating the watcher and download functionality and maybe we can tackle it later.